src/main/java/Course.java

```diff
@@ -1,47 +1,145 @@
+    private final Map<String, ArrayList<Session>> lectures;
+    private final Map<String, ArrayList<Session>> tutorials;
+
+    private static final String LEC = "LEC";
+    private static final String TUT = "TUT";
 
     /**
      * Constructs a course with an id and list of lecture and tutorial sessions
      *
      * @param courseId given id
-     * @param lectures given lectures
-     * @param tutorials given tutorials
      */
-    public Course(String courseId, ArrayList<Session> lectures, ArrayList<Session> tutorials) {
+    public Course(String courseId) {
         this.courseId = courseId;
-        this.lectures = lectures;
-        this.tutorials = tutorials;
+        this.lectures = new HashMap<String, ArrayList<Session>>();
+        this.tutorials = new HashMap<String, ArrayList<Session>>();
     }
 
+    /**
+     * Adds sessions to their respective sections in the hashmap.
+     *
+     * All sessions must have type and section.
+     *
+     * For example, a session that is a lecture has its type identified with Session.
+     */
+    public void addSessions(ArrayList<Session> sessions) {
+        String section;
+        String type;
+        Map<String, ArrayList<Session>> values;
+
+        for(Session s : sessions) {
+            type = s.getType();
+            section = s.getSection();
+
+            if (type == LEC) {
+                values = this.lectures;
+            } else if (type == TUT) {
+                values = this.tutorials;
+            } else {
+                continue;
+            }
+
+            if(values.containsKey(section)) {
+                values.get(section).add(s);
+            } else {
+                ArrayList<Session> temp = new ArrayList<>();
+                temp.add(s);
+                values.put(section, temp);
+            }
+
+            return;
+        }
+    }
+
+
     public String getCourseId() {
         return this.courseId;
     }
 
+    public String[] getLectureSections() {
+        return (String[]) this.lectures.keySet().toArray();
+    }
+
+    public String[] getTutorialSections() {
+        return (String[]) this.tutorials.keySet().toArray();
+    }
+
+
+    /** Returns a clone of the ArrayList of all lecture sessions in a given section.
+     * The getTutorials() method is very similar
+     *
+     * @param section the section code
+     * @return all lecture sessions in this section
+     */
+    public ArrayList<Session> getLectures(String section) {
+        if(lectures.containsKey(section)) {
+            return (ArrayList<Session>) lectures.get(section).clone();
+        } else {
+            return new ArrayList<Session>();
+        }
+    }
+
+    /** Returns the lecture sessions associated with the first key. If there is no first key,
+     * the map 'lectures' must be empty, and it returns an empty ArrayList.
+     *
+     * @return lecture sessions with the first section key in hashmap
+     */
     public ArrayList<Session> getLectures() {
```

Comment:
I think when you overload a function, all but the "base" function call should be as small as possible, and only do whatever work is needed to get the correct arguments to call the "base" function. The advantage is that you only really need to change the "base" function if u ever want to change how the function works.

So here it would be the call for `getLectures()` checks if the first key is there, and if it's there, it calls `getLectures(firstKey)`. Granted, if you do it this way, you're basically doing an extra "check if lectures contains firstKey" that you don't need, but that's trivial so IMO its fine.

Also remember to name variables & functions in `camelCase`, not `snake_case`

same thing for the `getTutorials` function, since it's basically same thing.

Reply:
Makes sense. I only included this method because its used in other classes that are implemented without section keys for now, like 'createBasicSchedule' in Scheduler. Once we fully refactor the code we will be able to remove tihs